### ISSUE TITLE
How to compute Gas + Uniswap Fees needed for a trade?

### ISSUE THREAD

**Issue by thegitparticle** [uniswap-integration/simple-uniswap-sdk#42]
After constructing a trade, how do I go about calculating gas fees + uniswap fees needed for a trade?

**Comment by LarryRyan0824** [comment]
Once you call the trade function of the UniswapPairFactory, the gas information will be in the object returned. You would have needed to provide the gas price in the UniswapPairSettings constructor. If you don't have enough balance for the input token, it will not be able to calculate it for you and will be undefined. 

It'll look similar to this. I made my own gas function since BSC wants to use the old gasGasPrice function

`// EIP-1559 supported chains use getFeeData
export async function getGasPriceForProvider(provider) {
    return [bscChainId].includes(provider._network.chainId)
        ? await provider?.getGasPrice()
        : (await provider?.getFeeData()).maxFeePerGas
}`

`let settings = new UniswapPairSettings({
            slippage,
            deadlineMinutes,
            disableMultihops: false,
            uniswapVersions,
            gasSettings: {
                getGasPrice: async () => formatUnits(await getGasPriceForProvider(provider), 'gwei')
            }
        })`

**Comment by thegitparticle** [comment]
Yes, this makes sense for gas. But how about the uniswap swap fee?
